src/HospitalLibrary/AnnualLeaves/AnnualLeaveService.cs

```diff
@@ -22,6 +28,18 @@ public IEnumerable<AnnualLeave> GetAllByDoctorId(int doctorId)
             return _unitOfWork.AnnualLeaveRepository.GetAllByDoctorId(doctorId);
         }
 
+        public IEnumerable<PendingRequestsDTO> GetAllPending()
+        {
+            List<PendingRequestsDTO> pendingRequests = new List<PendingRequestsDTO>();
+            foreach (AnnualLeave leave in _unitOfWork.AnnualLeaveRepository.GetAllPending())
+            {
+                Doctor doctor = _unitOfWork.DoctorRepository.GetOne(leave.DoctorId);
+                pendingRequests.Add(new PendingRequestsDTO(leave.Id, leave.DoctorId, doctor.Name + " " + doctor.Surname,
```

Comment:
Instead of passing all these arguments inside, you can pass only doctor and leave to dto constructor and do the logic inside.